akka-stream/src/main/scala/akka/stream/snapshot/MaterializerState.scala

```diff
@@ -41,11 +41,9 @@ object MaterializerState {
   @InternalApi
   private[akka] def requestFromSupervisor(supervisor: ActorRef)(
       implicit ec: ExecutionContext): Future[immutable.Seq[StreamSnapshot]] = {
-    // FIXME arbitrary timeout
+    // Arbitrary timeout: operation should always be quick, when it times out it will be because the materializer stopped
     implicit val timeout: Timeout = 10.seconds
-    (supervisor ? StreamSupervisor.GetChildren)
-      .mapTo[StreamSupervisor.Children]
-      .flatMap(msg => Future.sequence(msg.children.toVector.map(requestFromChild)))
+    (supervisor ? StreamSupervisor.GetChildrenSnapshots).mapTo[StreamSupervisor.ChildrenSnapshots].map(_.seq)
```

Comment:
ok, if all retry attempts fail that will send back a `Status.Failure`, which fail this outer Future